gto-support-db-coroutines/src/main/kotlin/org/ccci/gto/android/common/db/CoroutinesAsyncDao.kt

```diff
@@ -0,0 +1,24 @@
+package org.ccci.gto.android.common.db
+
+import android.database.sqlite.SQLiteDatabase
+import kotlinx.coroutines.async
+
+interface CoroutinesAsyncDao : CoroutinesDao {
+    fun <T : Any> findAsync(clazz: Class<T>, vararg key: Any) = coroutinesScope.async { find(clazz, *key) }
+    fun <T : Any> getAsync(query: Query<T>) = coroutinesScope.async { get(query) }
+    fun getCursorAsync(query: Query<*>) = coroutinesScope.async { getCursor(query) }
+
+    fun insertAsync(obj: Any) = insertAsync(obj, SQLiteDatabase.CONFLICT_NONE)
+    fun insertAsync(obj: Any, conflictAlgorithm: Int) = coroutinesScope.async { insert(obj, conflictAlgorithm) }
```

Comment:
I can see here that one actually calls the other. Is it to make it more flexible in the case you want to call it with a different ```confilctAlgorithm```?

Reply:
yes, it's to mirror the methods in Dao. The original AsyncDao was written before we used Kotlin, so there is probably some cleanup I can do here to utilize default params.

Reply:
`insertAsync` is the only one that qualifies for consolidation with a default parameter